java/system-test/system-test-dsl/src/test/java/sleeper/systemtest/dsl/testutil/drivers/InMemoryCompaction.java

```diff
@@ -160,8 +160,8 @@ private void finishJobs(SystemTestInstanceContext instance) {
             CompactionJobStatus status = jobTracker.getJob(job.getId()).orElseThrow();
             JobRunReport run = status.getRunsLatestFirst().stream().findFirst().orElseThrow();
             JobRunSummary summary = compact(job, tableProperties, instance.getStateStore(tableProperties), run);
-            jobTracker.jobFinished(job.finishedEventBuilder(summary).taskId(run.getTaskId()).build());
-            jobTracker.jobCommitted(job.committedEventBuilder(summary.getFinishTime().plus(Duration.ofMinutes(1))).taskId(run.getTaskId()).build());
+            jobTracker.jobFinished(job.finishedEventBuilder(summary).taskId(run.getTaskId()).jobRunId(job.getId()).build());
+            jobTracker.jobCommitted(job.committedEventBuilder(summary.getFinishTime().plus(Duration.ofMinutes(1))).taskId(run.getTaskId()).jobRunId(job.getId()).build());
```

Comment:
It looks like the job run ID isn't set on the replace file references request below, in the compact method. We could add validation for that, so that if the task ID is set the job run ID must be set, and vice versa.